pkgs/development/compilers/go/1.20.nix

```diff
@@ -141,13 +141,13 @@ stdenv.mkDerivation rec {
     # Contains the wrong perl shebang when cross compiling,
     # since it is not used for anything we can deleted as well.
     rm src/regexp/syntax/make_perl_groups.pl
-  '' + (if (stdenv.buildPlatform != stdenv.hostPlatform) then ''
+  '' + (if (stdenv.buildPlatform.system != stdenv.hostPlatform.system) then ''
     mv bin/*_*/* bin
     rmdir bin/*_*
     ${lib.optionalString (!(GOHOSTARCH == GOARCH && GOOS == GOHOSTOS)) ''
       rm -rf pkg/${GOHOSTOS}_${GOHOSTARCH} pkg/tool/${GOHOSTOS}_${GOHOSTARCH}
     ''}
-  '' else lib.optionalString (stdenv.hostPlatform != stdenv.targetPlatform) ''
+  '' else lib.optionalString (stdenv.hostPlatform.system != stdenv.targetPlatform.system) ''
     rm -rf bin/*_*
```

Comment:
Before this change `rm -rf bin/*_*` was included in the `preInstall` only for `x86_64-darwin`.

No idea why it was like that but it wasn't correct and it was a no op anyway as there wasn't anything for it to remove.

I think it's fine to merge this but I'd send to to staging so we aren't causing a pointless rebuild of all `x86_64-darwin` go packages on master.

Reply:
Thanks for figuring out what the difference is and confirming that it's fine to merge